caravel/models.py

```diff
@@ -644,6 +645,10 @@ def query(  # sqla
                 "Datetime column not provided as part table configuration "
                 "and is required by this type of chart"))
 
+        # Re-compile all the macros before rendering any SQL expression,
+        # so that we're sure everything is up-to-date
+        recompile_sql_macros()
```

Comment:
In a single worker deployment, recompiling on updates are the best strategy. But it doesn't work if we have multiple worker processes or even multiple server instances deployment (behind a load balancer).

To give you a bit more details, recompiling macros will create a jinja environment having all the macros in its global namespace. As you can see, if I call `recompile_sql_macros`, it will only recompile macros in one worker (whatever worker handles that call). If other workers executes a query, they will still use the old jinja environment.